exercises/tom_perkins/ex01_basics/src/counter.cpp

```diff
@@ -0,0 +1,87 @@
+#include "counter.h"
+
+/**
+ * Compares two (word, frequency) pairs to enable sorting.
+ * Sorting is done in descending order of usage i.e. the pair with higher frequency comes first.
+ */
+bool compareRecords(const std::pair<std::string, int>& x, const std::pair<std::string, int>& y)
+{
+	return x.second > y.second;
+}
+
+/**
+ * Writes a table of words and frequencies to supplied stream.
+ * Words are sorted in descending order of usage.
+ */
+void Counter::generateReport(std::ostream& stream)
```

Comment:
Good function naming. 

* Prefer to return something rather than modify the stream directly (I'll explain in detail in a bit)
* Shouldn't this function be const?